subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GrettySmokeTest.groovy

```diff
@@ -61,12 +62,19 @@ class GrettySmokeTest extends AbstractPluginValidatingSmokeTest {
 
         then:
         result.task(':checkContainerUp').outcome == SUCCESS
+
+        where:
+        grettyConfig << grettyConfigForCurrentJavaVersion()
     }
 
     @Override
     Map<String, Versions> getPluginsToValidate() {
         [
-            'org.gretty': Versions.of(TestedVersions.gretty)
+            'org.gretty': Versions.of(grettyConfigForCurrentJavaVersion().collect { it.version } as String[])
         ]
     }
+
+    static def grettyConfigForCurrentJavaVersion() {
+        TestedVersions.gretty.findAll { JavaVersion.current().isCompatibleWith(it.javaMinVersion as JavaVersion) }
```

Comment:
This would mean that you would test Gretty 3 on Java 11, because Java 11 is compatible with Java 8, right?

Reply:
Yes, I think it should work.